### PR TITLE
WI #1443 Add boolean retun value.

### PR DESCRIPTION
**Pull request by mayanje**

### CODE REVIEW

File: TypeCobol.Test/Report/ReportTestHelper.cs [hunk 1]
```diff
@@ -73,8 +74,17 @@ public static void ParseWithNodeListenerReportCompare(string fileName, string re
...
                         }
                     }
+                    else
+                    {
+                        return false;
+                    }
+                }
+                else
+                {
+                    return false;
```
Comment (fm-117) [inline TypeCobol.Test/Report/ReportTestHelper.cs]: The two blocks `else { return false; }` can be simplified as `return false;`at the end of the try block.

File: TypeCobol.Test/Report/ReportTestHelper.cs [hunk 2]
```diff
@@ -73,8 +74,17 @@ public static void ParseWithNodeListenerReportCompare(string fileName, string re
                             string result = sw.ToString();
                             string expected = File.ReadAllText(output, format.Encoding);
                             TypeCobol.Test.TestUtils.compareLines(input, result, expected, PlatformUtils.GetPathForProjectFile(output));
+                            return true;
                         }
                     }
+                    else
+                    {
+                        return false;
```
Comment (maxime645) [inline TypeCobol.Test/Report/ReportTestHelper.cs]: Can you return more information than just false in the case if there are diagnostics or the report is null ?
Running the test in debug just to know what happened can be prevented if the information is in the output.

Comment (mayanje) [reply TypeCobol.Test/Report/ReportTestHelper.cs]: Do you want to return the Diagsnotics list?

File: TypeCobol.Test/Report/ReportTestHelper.cs (hunk 1 above)
Comment (mayanje) [reply TypeCobol.Test/Report/ReportTestHelper.cs]: Just a matter of style

File: TypeCobol.Test/Report/ReportTestHelper.cs (hunk 2 above)
Comment (maxime645) [reply TypeCobol.Test/Report/ReportTestHelper.cs]: Not necessarily the Diagnostics list, but at least a message explaining why the test failed (e.g. "diagnostics encountered", "report not generated correctly"...)

Comment (mayanje) [reply TypeCobol.Test/Report/ReportTestHelper.cs]: Is a return code enough?

**Pull request merged**